torch/_inductor/codegen/cuda/cuda_kernel.py

```diff
@@ -157,25 +166,55 @@ def call_kernel(
         as well as all required inputs and outputs.
         """
         wrapper = V.graph.wrapper_code
-        _, call_args, _, arg_types = self.args.python_argdefs()
+
+        if V.graph.cpp_wrapper:
+            # Make sure we initialize these kernels since they're exported as
+            # C-style symbol names.
+            assert isinstance(wrapper, CppWrapperCpu)
+            wrapper.initialized_kernels[name] = self
+            # Kinda hacky because we always originally initialize name with "KERNEL_NAME"
+            # So, we replace with the real kernel name passed as an arg to this function.
+            self.signature = self.signature.replace("KERNEL_NAME", name)
+
+        # NOTE:
+        # We have to use python_argdefs even for cpp wrapper because python_argdefs
+        # gives us torch.dtype which is useful when calling
+        # generate_args_decl(call_args, arg_types) in generate_kernel_call()
+
+        if V.graph.cpp_wrapper:
```

Comment:
Hmm, I feel like our convention is that we don't diverge with respect to cpp_wrapper in cuda_kernel.py. Instead, we rely on calling certain wrapper methods to differentiate cpp and non-cpp codegens. 

Adding @desertfire for more feedback.

Reply:
@chenyang78 made a valid point. If the logic is complex, the convention is to separate them in the two wrapper codegen classes.